lessons/lesson-3/tasks/task-1.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Замыкания могут применяться для простой реализации инкапсуляции, т.е. для того чтобы скрыть реализацию
+ * какой-то функциональности.
+ *
+ * Копилка
+ *
+ * Реализуйте функцию MoneyBox, который будет создавать объект с двумя методами:
+ * 1. addCoin - Добавляет 1 монету в копилку
+ * 2. getAmount - Возвращает количество монет в копилке
+ *
+ * Не должно быть возможности задать количество монет напрямую, минуя метод addCoin. При создании в копилке нет монет.
+ *
+ * Пример:
+ * const box = new MoneyBox();
+ * box.getAmount(); // 0
+ * box.addCoin();
+ * box.addCoin();
+ * box.getAmount(); // 2
+ */
+
+const MoneyBox = function() {
+	let coins = 0
+	this.addCoin = () => {
+		coins += 1
+	}
+	this.getAmount = () => {
+		return coins
+	}
```

Comment:
Однострочные стрелочные функции можно записать компактнее:

```suggestion
	this.addCoin = () => ++coins
	this.getAmount = () => coins
```

Reply:
@PonomareVlad В вашем варианте кода на 23 строчке будет возвращено значение `coins` после выполнения метода `addCoin`. В моём варианте возвращается `undefined`. Можно ли как-то в одну строчку сделать так, чтобы тоже возвращалось значение `undefined`? Другой вопрос: имеет ли смысл в данной ситуации вообще возвращать значение `undefined`?

Reply:
Зависит от вашей реализации 🙂

Иногда удобно получить сразу результат вычисления, без дополнительного обращения к переменной или вызова функции

Однако, если вы хотите возвращать именно `undefined`, то можно сделать так:

```js
this.addCoin = () => (++coins, undefined)
```

[Подробности](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Operators/Comma_Operator)

P.S. Не рекомендую часто использовать этот трюк, т.к. это сильно снижает читабельность кода + некоторые команды (компании) будут против подобного стиля написания кода и попросят переписать это так, как вы написали изначально (используя мультистрочный формат)

Reply:
@PonomareVlad Исправил.